pkg/functionconfig/reader.go

```diff
@@ -66,6 +66,12 @@ func (r *Reader) Read(reader io.Reader, configType string, config *Config) error
 		}
 	}
 
+	// if there is an http trigger in the codeEntry config remove default from config
+	if len(GetTriggersByKind(codeEntryConfig.Spec.Triggers, "http")) > 0 {
+		defaultHTTPTrigger := GetDefaultHTTPTrigger()
+		delete(config.Spec.Triggers, defaultHTTPTrigger.Name)
+	}
```

Comment:
This assume it has 1 custom HTTP trigger + the default one
what if the file has 3 custom http triggers?
what is the file has 1 customer http trigger, and the `config` has a different one?

At the very least this needs to have these:
- Make validation on the CE config before merge to check for: custom + default (fail), more than one custom (fail), I guess we can allow no http triggers here
- Make a validation on the merged config for: custom + default (can remedy like you do here), more than one custom (fail), no http triggers - (dunno how this is possible but can remedy I guess)